src/MongoDB.Driver.Core/Core/WireProtocol/CommandWireProtocol.cs

```diff
@@ -102,25 +118,31 @@ private IWireProtocol<TCommandResult> CreateCommandUsingCommandMessageWireProtoc
                 _readPreference,
                 _databaseNamespace,
                 _command,
+                _commandPayloads,
                 _commandValidator,
                 _additionalOptions,
                 _responseHandling,
                 _resultSerializer,
-                _messageEncoderSettings);
+                _messageEncoderSettings,
+                _postWriteAction);
         }
 
         private IWireProtocol<TCommandResult> CreateCommandUsingQueryMessageWireProtocol()
         {
+            var responseHandling = _responseHandling == CommandResponseHandling.NoResponseExpected ? CommandResponseHandling.Ignore : _responseHandling;
```

Comment:
Since OP_QUERY doesn't support anything like MoreToCome we map NoResponseExpected to Ignore.